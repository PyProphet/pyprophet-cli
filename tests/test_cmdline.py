# encoding: utf-8
from __future__ import print_function
import pdb
import subprocess
import pytest
import os
import shutil
import collections
import tempfile

import pandas

pandas.options.display.precision=6
pandas.options.display.width=200


@pytest.fixture(scope="session")
def test_data_folder():
    here = os.path.dirname(os.path.abspath(__file__))
    return os.path.join(here, "data")


@pytest.fixture(scope="session", params=[1, 3])
def setup(test_data_folder, request):
    num_files = request.param
    tmpdir = tempfile.mkdtemp()
    data_folder = os.path.join(tmpdir, "data_folder")
    os.makedirs(data_folder)
    working_folder = os.path.join(tmpdir, "working_folder")
    for i in range(num_files):
        shutil.copy(os.path.join(test_data_folder, "test_data.txt"), os.path.join(data_folder, "data_%d.txt" % i))

    Setup = collections.namedtuple("Setup", "data_folder working_folder number_input_files")
    return Setup(data_folder, working_folder, num_files)


def test_check(setup):
    cmd = ("pyprophet-brutus check --data-folder %s" % setup.data_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0


def test_subsample(setup):
    cmd = ("pyprophet-brutus subsample --job-number 1 --job-count 1 "
           "--sample-factor 0.1 "
           "--random-seed 43 "
           "--ignore-invalid-scores "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s") % (setup.data_folder, setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
    files = os.listdir(setup.working_folder)
    assert len(files) == setup.number_input_files + 1

    files.sort()
    subsamples = pandas.read_csv(os.path.join(setup.working_folder, files[-1]), sep="\t")
    assert subsamples.shape == (934, 21)

    cmd = ("pyprophet-brutus subsample --job-number 1 --job-count 2 "
           "--sample-factor 0.1 "
           "--random-seed 43 "
           "--ignore-invalid-scores "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s "
           "--local-folder %s") % (setup.data_folder, setup.working_folder, tempfile.mkdtemp())
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
    files = os.listdir(setup.working_folder)
    assert len(files) == 2 if setup.number_input_files == 1 else 3

    files.sort()
    subsamples = pandas.read_csv(os.path.join(setup.working_folder, files[-1]), sep="\t")
    assert subsamples.shape == (934, 21)

    cmd = ("pyprophet-brutus subsample --job-number 1 --job-count 2 "
           "--sample-factor 0.1 "
           "--random-seed 43 "
           "--ignore-invalid-scores "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s") % (setup.data_folder, setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
    files = os.listdir(setup.working_folder)
    assert len(files) == 2 if setup.number_input_files == 1 else 3

    files.sort()

    subsamples = pandas.read_csv(os.path.join(setup.working_folder, files[-1]), sep="\t")
    assert subsamples.shape == (934, 21)


def test_learn(setup, regtest):
    cmd = ("pyprophet-brutus learn --random-seed 43 --working-folder %s" % setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0

    files = sorted(os.listdir(setup.working_folder))

    for i, file_ in enumerate(files):
        path = os.path.join(setup.working_folder, file_)
        print(i, "%7d" % os.stat(path).st_size, file_, file=regtest)

    for name in ("weights.txt", "sum_stat_subsampled.txt"):
        df = pandas.read_csv(os.path.join(setup.working_folder, name), header=None, sep="\t")
        print(df, file=regtest)
        # print(open(os.path.join(setup.working_folder, name), "r").read(), file=regtest)


def test_score(setup, regtest):
    cmd = ("pyprophet-brutus apply_weights --job-number 1 --job-count 1 "
           "--data-filename-pattern '*.txt' "
           "--data-folder %s --working-folder %s") % (setup.data_folder, setup.working_folder)
    ret_code = subprocess.call(cmd, shell=True)
    assert ret_code == 0
