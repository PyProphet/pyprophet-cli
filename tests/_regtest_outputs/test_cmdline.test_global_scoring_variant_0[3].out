 0      11 work_folder/extra_group_columns.txt
 1    6518 work_folder/full_stat_subsampled.txt
 2      17 work_folder/invalid_columns.txt
 3     392 work_folder/score_columns.txt
 4     375 work_folder/score_columns_in_order_from_classifier.txt
 5 1107656 work_folder/subsampled_data_0.txt
 6 1107656 work_folder/subsampled_data_1.txt
 7 1107656 work_folder/subsampled_data_2.txt
 8     706 work_folder/sum_stat_subsampled.txt
 9     433 work_folder/weights.txt

weights
            0
0    0.205337
1   -0.113892
2   -3.210055
3    1.238617
4    2.275234
5   -2.015747
6   -0.577659
7  -11.213477
8    0.115411
9    0.009374
10  -0.006875
11  -8.796926
12   0.304486
13  -0.561132
14   4.125067
15   2.151725
16   0.117744
   qvalue    svalue        pvalue   TP  FP         TN          FN           FDR      sens    cutoff
0    0.00  0.015385  4.341387e-10    9   0  36.185567  539.814433  1.860595e-09  0.016399  6.334185
1    0.01  0.986601  1.365491e-01  541   5  31.185567    7.814433  9.042985e-03  0.985761  1.495220
2    0.02  1.000000  3.114364e-01  550  11  25.185567   -1.185567  2.009267e-02  1.000000  0.601479
3    0.05  1.000000  8.370135e-01  552  30   6.185567   -3.185567  5.198842e-02  1.000000 -0.246746
4    0.10  1.000000  9.149512e-01  552  33   3.185567   -3.185567  5.659492e-02  1.000000 -1.182657
5    0.20       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN
6    0.30       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN
7    0.40       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN
8    0.50       NaN           NaN  NaN NaN        NaN         NaN           NaN       NaN       NaN 0   89146 work_folder/data_0_score_data.npz
 1 2162728 work_folder/data_0_top_score_data.hdf
 2   89146 work_folder/data_1_score_data.npz
 3 2162728 work_folder/data_1_top_score_data.hdf
 4   89146 work_folder/data_2_score_data.npz
 5 2162728 work_folder/data_2_top_score_data.hdf
 6      11 work_folder/extra_group_columns.txt
 7    6518 work_folder/full_stat_subsampled.txt
 8      17 work_folder/invalid_columns.txt
 9     392 work_folder/score_columns.txt
10     375 work_folder/score_columns_in_order_from_classifier.txt
11 1107656 work_folder/subsampled_data_0.txt
12 1107656 work_folder/subsampled_data_1.txt
13 1107656 work_folder/subsampled_data_2.txt
14     706 work_folder/sum_stat_subsampled.txt
15     433 work_folder/weights.txt
 0 2930213 result_folder_3/data_0_scored.txt
 1 2930213 result_folder_3/data_1_scored.txt
 2 2930213 result_folder_3/data_2_scored.txt
 3   29734 result_folder_3/report_grouped_by_peptide_id.pdf
 4   29734 result_folder_3/report_grouped_by_transition_group_id.pdf
 5    1164 result_folder_3/summary_stats_grouped_by_peptide_id.txt
 6    1164 result_folder_3/summary_stats_grouped_by_transition_group_id.txt
 0   89146 work_folder/data_0_score_data.npz
 1 2162728 work_folder/data_0_top_score_data.hdf
 2   89146 work_folder/data_1_score_data.npz
 3 2162728 work_folder/data_1_top_score_data.hdf
 4   89146 work_folder/data_2_score_data.npz
 5 2162728 work_folder/data_2_top_score_data.hdf
 6      11 work_folder/extra_group_columns.txt
 7    6518 work_folder/full_stat_subsampled.txt
 8      17 work_folder/invalid_columns.txt
 9     392 work_folder/score_columns.txt
10     375 work_folder/score_columns_in_order_from_classifier.txt
11 1107656 work_folder/subsampled_data_0.txt
12 1107656 work_folder/subsampled_data_1.txt
13 1107656 work_folder/subsampled_data_2.txt
14     706 work_folder/sum_stat_subsampled.txt
15     433 work_folder/weights.txt

num_null   : 25.00
num_total  : 387.00
stats shape: (387, 10)
mean top target scores: 4.367880
sdev top target scores: 1.518137
mean top decoy  scores: 0.000000
sdev top decoy  scores: 1.000000

   qvalue    svalue    pvalue   TP  FP            TN            FN       FDR      sens    cutoff
0    0.00  0.872915  0.000186  316   0  2.500000e+01  4.600000e+01  0.000212  0.872928  3.559014
1    0.01  0.998093  0.147618  361   4  2.100000e+01  1.000000e+00  0.010111  0.997238  1.046707
2    0.02  1.000000  0.282889  361   7  1.800000e+01  1.000000e+00  0.019218  0.997238  0.574281
3    0.05  1.000000  0.795019  363  20  5.000000e+00 -1.000000e+00  0.051894  1.000000 -0.823961
4    0.10  1.000000  0.982481  362  25 -3.552714e-15  3.552714e-15  0.063468  1.000000 -2.107911
5    0.20       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
6    0.30       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
7    0.40       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
8    0.50       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN

num_null   : 25.00
num_total  : 387.00
stats shape: (387, 10)
mean top target scores: 4.367880
sdev top target scores: 1.518137
mean top decoy  scores: 0.000000
sdev top decoy  scores: 1.000000

   qvalue    svalue    pvalue   TP  FP            TN            FN       FDR      sens    cutoff
0    0.00  0.872915  0.000186  316   0  2.500000e+01  4.600000e+01  0.000212  0.872928  3.559014
1    0.01  0.998093  0.147618  361   4  2.100000e+01  1.000000e+00  0.010111  0.997238  1.046707
2    0.02  1.000000  0.282889  361   7  1.800000e+01  1.000000e+00  0.019218  0.997238  0.574281
3    0.05  1.000000  0.795019  363  20  5.000000e+00 -1.000000e+00  0.051894  1.000000 -0.823961
4    0.10  1.000000  0.982481  362  25 -3.552714e-15  3.552714e-15  0.063468  1.000000 -2.107911
5    0.20       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
6    0.30       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
7    0.40       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN
8    0.50       NaN       NaN  NaN NaN           NaN           NaN       NaN       NaN       NaN

transition_group_id	run_id	decoy	main_var_xx_swath_prelim_score	var_bseries_score	var_elution_model_fit_score	var_intensity_score	var_isotope_correlation_score	var_isotope_overlap_score	var_library_corr	var_library_rmsd	var_log_sn_score	var_massdev_score	var_massdev_score_weighted	var_norm_rt_score	var_xcorr_coelution	var_xcorr_coelution_weighted	var_xcorr_shape	var_xcorr_shape_weighted	var_yseries_score	var_faulty_score	peptide_id	peak_group_rank	d_score	transition_group_id_m_score	transition_group_id_p_value	peptide_id_m_score	peptide_id_p_value

459_run0	0	0	5.27893972397	5.0	0.993098139763	0.930480718613	0.998724222183	0.0	0.912406027317	0.0398625209928	5.01656723022	8.83918857574	3.88453722	0.00229200161994	0.0	0.0	0.960350096226	0.970979332924	6.0		459_run0	1	5.00816345215	0.000211860664422	2.74759628383e-07	0.000211860664422	2.74759628383e-07


transition_group_id	run_id	decoy	main_var_xx_swath_prelim_score	var_bseries_score	var_elution_model_fit_score	var_intensity_score	var_isotope_correlation_score	var_isotope_overlap_score	var_library_corr	var_library_rmsd	var_log_sn_score	var_massdev_score	var_massdev_score_weighted	var_norm_rt_score	var_xcorr_coelution	var_xcorr_coelution_weighted	var_xcorr_shape	var_xcorr_shape_weighted	var_yseries_score	var_faulty_score	peptide_id	peak_group_rank	d_score	transition_group_id_m_score	transition_group_id_p_value	peptide_id_m_score	peptide_id_p_value

459_run0	0	0	5.27893972397	5.0	0.993098139763	0.930480718613	0.998724222183	0.0	0.912406027317	0.0398625209928	5.01656723022	8.83918857574	3.88453722	0.00229200161994	0.0	0.0	0.960350096226	0.970979332924	6.0		459_run0	1	5.00816345215	0.000211860664422	2.74759628383e-07	0.000211860664422	2.74759628383e-07


transition_group_id	run_id	decoy	main_var_xx_swath_prelim_score	var_bseries_score	var_elution_model_fit_score	var_intensity_score	var_isotope_correlation_score	var_isotope_overlap_score	var_library_corr	var_library_rmsd	var_log_sn_score	var_massdev_score	var_massdev_score_weighted	var_norm_rt_score	var_xcorr_coelution	var_xcorr_coelution_weighted	var_xcorr_shape	var_xcorr_shape_weighted	var_yseries_score	var_faulty_score	peptide_id	peak_group_rank	d_score	transition_group_id_m_score	transition_group_id_p_value	peptide_id_m_score	peptide_id_p_value

459_run0	0	0	5.27893972397	5.0	0.993098139763	0.930480718613	0.998724222183	0.0	0.912406027317	0.0398625209928	5.01656723022	8.83918857574	3.88453722	0.00229200161994	0.0	0.0	0.960350096226	0.970979332924	6.0		459_run0	1	5.00816345215	0.000211860664422	2.74759628383e-07	0.000211860664422	2.74759628383e-07

